mmf/common/test_reporter.py

```diff
@@ -151,7 +151,14 @@ def add_to_report(self, report, model):
         if not is_master():
             return
 
-        results = self.current_dataset.format_for_prediction(report)
+        inference_type = None
+        if self.config.model == "butd":
+            inference_type = self.config.model_config.butd.inference.type
+
+        if inference_type in ["beam_search", "nucleus_sampling"]:
+            results = self.current_dataset.format_for_caption_generation(report)
+        else:
+            results = self.current_dataset.format_for_prediction(report)
```

Comment:
Let's keep only the `format_for_prediction` and remove this logic here. If report contains `captions` use that in `format_for_prediction` otherwise fall back to old implementation.

Reply:
I agree, though I think a potential problem is that "caption" is an overloaded field, therefore it seems necessary to check for the inference type, and I am not sure if we can easily check that `model_config` in the dataset builders..